builder/ibmcloud/vpc/config.go

```diff
@@ -85,21 +86,46 @@ func (c *Config) Prepare(raws ...interface{}) ([]string, error) {
 		errs = packer.MultiErrorAppend(errs, errors.New("a subnet_id must be specified"))
 	}
 
-	if (c.CatalogOfferingCRN != "" || c.CatalogOfferingVersionCRN != "") && (c.VSIBaseImageID != "" || c.VSIBaseImageName != "") && c.VSIBootVolumeID != "" {
-		errs = packer.MultiErrorAppend(errs, errors.New("only one of (vsi_base_image_id or vsi_base_image_name) or (catalog_offering_crn or catalog_offering_version_crn) or vsi_boot_volume_id is required"))
-	} else if (c.CatalogOfferingCRN != "" || c.CatalogOfferingVersionCRN != "") && (c.VSIBaseImageID != "" || c.VSIBaseImageName != "") {
-		errs = packer.MultiErrorAppend(errs, errors.New("only one of (vsi_base_image_id or vsi_base_image_name) or (catalog_offering_crn or catalog_offering_version_crn) is required"))
-	} else if (c.CatalogOfferingCRN != "" || c.CatalogOfferingVersionCRN != "") && c.VSIBootVolumeID != "" {
-		errs = packer.MultiErrorAppend(errs, errors.New("only one of (catalog_offering_crn or catalog_offering_version_crn) or vsi_boot_volume_id is required"))
-	} else if (c.VSIBaseImageID != "" || c.VSIBaseImageName != "") && c.VSIBootVolumeID != "" {
-		errs = packer.MultiErrorAppend(errs, errors.New("only one of (vsi_base_image_id or vsi_base_image_name) or vsi_boot_volume_id is required"))
-	} else if c.CatalogOfferingCRN != "" && c.CatalogOfferingVersionCRN != "" {
-		errs = packer.MultiErrorAppend(errs, errors.New("only one of (catalog_offering_crn or catalog_offering_version_crn) is required"))
-	} else if c.VSIBaseImageID != "" && c.VSIBaseImageName != "" {
-		errs = packer.MultiErrorAppend(errs, errors.New("only one of (vsi_base_image_id or vsi_base_image_name) is required"))
-	} else if c.CatalogOfferingCRN == "" && c.CatalogOfferingVersionCRN == "" && c.VSIBaseImageID == "" && c.VSIBaseImageName == "" && c.VSIBootVolumeID == "" {
-		errs = packer.MultiErrorAppend(errs, errors.New("one of (vsi_base_image_id or vsi_base_image_name) or (catalog_offering_crn or catalog_offering_version_crn) or vsi_boot_volume_id  is required"))
+	var oneOfInput int // validation for mutually exclusive fields.
+
+	if c.VSIBaseImageID != "" {
+		oneOfInput = oneOfInput + 1
+	}
+	if c.VSIBaseImageName != "" {
+		oneOfInput = oneOfInput + 1
+	}
+	if c.CatalogOfferingCRN != "" {
+		oneOfInput = oneOfInput + 1
+	}
+	if c.CatalogOfferingVersionCRN != "" {
+		oneOfInput = oneOfInput + 1
+	}
+	if c.VSIBootVolumeID != "" {
+		oneOfInput = oneOfInput + 1
 	}
+	if c.VSIBootSnapshotID != "" {
+		oneOfInput = oneOfInput + 1
+	}
+
+	if oneOfInput != 1 {
+		errs = packer.MultiErrorAppend(errs, errors.New("only one of (vsi_base_image_id or vsi_base_image_name) or (catalog_offering_crn or catalog_offering_version_crn) or vsi_boot_volume_id or vsi_boot_snapshot_id is required"))
+	}
+
+	// if (c.CatalogOfferingCRN != "" || c.CatalogOfferingVersionCRN != "") && (c.VSIBaseImageID != "" || c.VSIBaseImageName != "") && c.VSIBootVolumeID != "" && c.VSIBootSnapshotID != "" {
```

Comment:
Can you please remove this commented lines if not required..

Thank you

Reply:
Logic changes, seeking review from other dev stake holders on this, will clear ones done.

Reply:
removed